### PR TITLE
Change Tests to Work with New Generator Script

### LINKED ISSUE #712: Change expected test data for e2e tests
**Description**:
As described in https://github.com/ral-facilities/datagateway-api/issues/236, the data on preprod will need to be changed in line with the fixes on the generator script to ensure consistent data is generated every single time. Before that is done however, the e2e tests should be edited so they work with the new version of the stable generated by the script. 

I'll branch off my JSON casing branch and make these changes once I've fixed the script.

**Acceptance criteria**:
- [ ] e2e tests should pass when using data generated from `icat_db_generator.py`

### PR DESCRIPTION
**Pull request by MRichards99**
## Description
This fixes #712. These changes make the e2e tests work with the updated icatdb generator script, changed as per https://github.com/ral-facilities/datagateway-api/issues/236. I've tried to use similar test data for each file as I noticed trends between files before these changes. For example, I've used `/browse/instrument/1/facilityCycle/16/investigation` quite a bit across different files.

I've also changed how the e2e tests for the admin download table work (https://github.com/ral-facilities/datagateway/commit/f121091181ced7564fa7808ff00c35e540089086 and https://github.com/ral-facilities/datagateway/commit/bef1f54f2ec7d41f0e1228eb1902607e1999a907). When looking for a prepared ID, it matches to a regex pattern instead of looking for a specific UUID as these will change for each ICAT instance. I've also tried to reduce the reliance on checking for prepared IDs (though not eliminated them) as I've noticed that even on fresh instances these can sometimes appear to be blank.

As preparation for the local CI work I've been doing, I've removed references to preprod URLs and now instead get the URLs for the settings files. There were some references not relating to e2e tests so I've also made these references look at the settings file. I've also done the same for a few facility names (mostly in `commands.js`) where I noticed relevant TODOs. I haven't removed all references to LILS just to avoid breaking things.

## Testing instructions
These have been tested while I've been working on the local CI, as can be seen passing [here](https://github.com/ral-facilities/datagateway/actions/runs/1037213161). To test these locally, you should repopulate your icatdb with `icatdb_generator.py` on master of DataGateway API. There are then a number of changes you need to make to your ICAT instance if not already done:

- Add `anon/anon` to rootUserNames in ICAT Server's `run.properties`
-  Disable two level storage on IDS by commenting out everything from the `# Properties for archive storage` section in `run.properties`, specifically: `plugin.archive.class`, `plugin.archive.dir`, `writeDelaySeconds`, `startArchivingLevel1024bytes`, `stopArchivingLevel1024bytes`, `storageUnit` and `tidyBlockSize`
- Reinstall ICAT Server and IDS Server after these changes
- Disable globus using `/topcat/admin/downloadType/globus/status`, using the message `"GLOBUS has been disabled for testing"`
- Recreate Lucene indexes for Datafile, Dataset and Investigation using `/icat/lucene/db/[ENTITY-NAME]/0`

Preprod's icatdb will need to be dropped and data regenerated using the fixed script.  There's a rough guide for it in one of the [API's CI jobs](https://github.com/ral-facilities/datagateway-api/blob/master/.github/workflows/ci-build.yml#L217). It would be great to get a dump (with the `--skip-comments` option) so I could check the generator script against that dump on the API's CI. I'm guessing this is something that @louise-davies and I will coordinate to do, but just writing here so I don't forget.

- [ ] Review code
- [ ] Check Actions build (won't be relevant until preprod data has been changed or my local CI PR has been merged into this branch)
- [ ] Review changes to test coverage
- [ ] Could someone look over the non-test files to make sure I haven't done anything silly? 

## Agile board tracking
connect to #712

### CODE REVIEW

File: packages/datagateway-download/src/__mocks__/axios.ts [hunk 1]
```diff
@@ -10,19 +13,19 @@ const requests = {
       return Promise.resolve({
         data: {
           facilityName: 'LILS',
```
Comment (sam-glendenning) [inline packages/datagateway-download/src/__mocks__/axios.ts:15]: Could this be `settings.facilityName` ?

File: packages/datagateway-search/cypress/integration/search/datafileSearch.spec.ts [hunk 2]
```diff
@@ -89,10 +92,10 @@ describe('Datafile search tab', () => {
       .should('not.exist');
   });
 
-  it('should link to a parent dataset', () => {
+  it.only('should link to a parent dataset', () => {
```
Comment (sam-glendenning) [inline packages/datagateway-search/cypress/integration/search/datafileSearch.spec.ts:95]: This seems to have sneaked in somehow. This is my fault, I was working on this a short while ago.

**Comment by sam-glendenning** [comment]
There's still quite a few places where `LILS` is hard coded. Could this be replaced with `settings.facilityName`?

File: packages/datagateway-dataview/cypress/integration/table/dls/datafiles.spec.ts [hunk 3]
```diff
@@ -28,10 +28,11 @@ describe('DLS - Datafiles Table', () => {
   it('should be able to scroll down and load more rows', () => {
     cy.get('[aria-rowcount="50"]').should('exist');
     cy.get('[aria-label="grid"]').scrollTo('bottom');
-    cy.get('[aria-rowcount="56"]').should('exist');
+    cy.get('[aria-rowcount="55"]').should('exist');
   });
 
-  it('should be able to resize a column', () => {
+  // TODO - Fix test which doesn't pass on local CI - functionality does work
+  it.skip('should be able to resize a column', () => {
```
Comment (sam-glendenning) [inline packages/datagateway-dataview/cypress/integration/table/dls/datafiles.spec.ts:35]: Simply a note for others but this has been fixed in https://github.com/ral-facilities/datagateway/pull/756, it just hasn't been merged yet.

**Comment by sam-glendenning** [comment]
I'm happy to make the necessary fixes here once I've heard back from Matt, plus will also merge in master once again to get this fully up to date before approving its merge.

File: packages/datagateway-download/src/__mocks__/axios.ts (hunk 1 above)
Comment (MRichards99) [reply packages/datagateway-download/src/__mocks__/axios.ts:15]: I didn't remove all instances of the hardcoded LILS as I didn't want to break anything but it's probably a good idea if the settings file has already been read in above

File: packages/datagateway-download/cypress/integration/adminDownloadStatus.spec.ts [hunk 4]
```diff
@@ -35,64 +35,67 @@ describe('Admin Download Status', () => {
...
-      '675eea67-bad7-4209-b54e-87f0d97eb50d'
-    );
+
+    // Typical `.should('match'...)`  doesn't work for text, tries to match the element,
+    // hence a slightly different approach for doing regex on the actual text
+    cy.get('[aria-rowindex="1"] [aria-colindex="4"]').find('p').should(($preparedId) => {
+      expect($preparedId[0].textContent).match(
+        /[0-9a-zA-Z]{8}\-[0-9a-zA-Z]{4}\-[0-9a-zA-Z]{4}\-[0-9a-zA-Z]{4}\-[0-9a-zA-Z]{12}/
+      );
+    });
```
Comment (sam-glendenning) [inline packages/datagateway-download/cypress/integration/adminDownloadStatus.spec.ts:45]: I may be wrong but could we use `.contains('expected text')` here?

File: packages/datagateway-dataview/cypress/integration/table/dls/datafiles.spec.ts (hunk 3 above)
Comment (MRichards99) [reply packages/datagateway-dataview/cypress/integration/table/dls/datafiles.spec.ts]: I've just merged this so the skip should no longer be present

**Review by sam-glendenning — APPROVED** [review]
Finally ready to go! Great job, Matt, this all looks good to me

**Pull request merged**